src/main/java/io/temporal/samples/dsl/models/ActivityInvocation.java

```diff
@@ -0,0 +1,55 @@
+package io.temporal.samples.dsl.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.workflow.ActivityStub;
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+import java.util.Map;
+
+public class ActivityInvocation {
+  public String name;
+  public String method;
+  public String[] arguments;
+  public String result;
+
+  @JsonCreator
+  public ActivityInvocation(
+      @JsonProperty("name") String name,
+      @JsonProperty("method") String method,
+      @JsonProperty("arguments") String[] arguments,
+      @JsonProperty("result") String result) {
+    this.name = name;
+    this.arguments = arguments;
+    this.result = result;
+    this.method = method;
+  }
+
+  public Void execute(Map<String, String> bindings) {
+    String[] args = makeInput(this.arguments, bindings);
+    ActivityStub stub =
+        Workflow.newUntypedActivityStub(
+            ActivityOptions.newBuilder()
+                .setStartToCloseTimeout(Duration.ofMinutes(5))
+                .setTaskQueue("dsl")
+                .build());
+
+    String results = stub.execute(this.method, String.class, new Object[] {args});
```

Comment:
It would be nice to change activities to accept the args as the current implementation doesn't use them at all.
I'm not sure why new Object[] {args} is needed instead of just args.

Reply:
If you don't do this, you get a warning in java.  The method takes a list of args which need to be expanded.  This was the workaround I found on stack-overflow to get around it.

`It would be nice to change activities to accept the args as the current implementation doesn't use them at all.`

Doesn't passing in the args like that use it in the activities?  When I look at the stack trace, it looks like it's using it as inputs in the payload.  Were you referring to something else?

![image](https://user-images.githubusercontent.com/16171726/94376048-de6dd380-00cc-11eb-947f-5b08044e4185.png)